pkg/api/vlabs/validate.go

```diff
@@ -635,24 +634,11 @@ func (a *Properties) validateAddons() error {
 				}
 			case "nvidia-device-plugin":
 				if to.Bool(addon.Enabled) {
-					version := common.RationalizeReleaseAndVersion(
-						a.OrchestratorProfile.OrchestratorType,
-						a.OrchestratorProfile.OrchestratorRelease,
-						a.OrchestratorProfile.OrchestratorVersion,
-						false,
-						false)
-					if version == "" {
-						return errors.Errorf("the following user supplied OrchestratorProfile configuration is not supported: OrchestratorType: %s, OrchestratorRelease: %s, OrchestratorVersion: %s. Please check supported Release or Version for this build of aks-engine", a.OrchestratorProfile.OrchestratorType, a.OrchestratorProfile.OrchestratorRelease, a.OrchestratorProfile.OrchestratorVersion)
-					}
-					sv, err := semver.Make(version)
-					if err != nil {
-						return errors.Errorf("could not validate version %s", version)
-					}
-					minVersion, err := semver.Make("1.10.0")
+					isValidVersion, err := common.IsValidMinVersion(a.OrchestratorProfile.OrchestratorType, a.OrchestratorProfile.OrchestratorVersion, "1.10.0")
```

Comment:
this change of logic might be breaking here. In some cases, validate actually runs on the apimodel before default values are set and `OrchestratorRelease` might be set but not `OrchestratorVersion`, hence the previous logic using `RationalizeReleaseAndVersion` with whatever is available: OrchestratorRelease, OrchestratorVersion, or neither (which is also valid, in that case the default is picked).

Reply:
makes sense, i made some changes. let me if this works